multimap/src/main/java/org/infinispan/multimap/impl/SetBucket.java

```diff
@@ -107,20 +111,32 @@ public String toString() {
       return "SetBucket{values=" + Util.toStr(values) + '}';
    }
 
-   public boolean add(V value) {
-      return values.add(new MultimapObjectWrapper<>(value));
+   public SetBucketResult<Boolean, V> addAll(Collection<V> values) {
+      Set<MultimapObjectWrapper<V>> existing = new HashSet<>(this.values);
+      boolean added = existing.addAll(values.stream().map(MultimapObjectWrapper::new).collect(Collectors.toSet()));
```

Comment:
This technically would be slightly better to just do a for loop on the values and call add directly as we don't need to create the intermediate Set, but I am fine if we don't change it.